mgmtsystem/google.py

```diff
@@ -0,0 +1,312 @@
+            return True
+
+        self.logger.info("Deleting Google Cloud instance {}".format(instance_name))
+        operation = self._instances.delete(
+            project=self._project, zone=self._zone, instance=instance_name).execute()
+        wait_for(lambda: self._nested_wait_vm_running(operation['name']), delay=0.5,
+            num_sec=timeout, message="Delete {}".format(instance_name))
+        return True
+
+    def restart_vm(self, instance_name):
+        self.logger.info("Restarting Google Cloud instance {}".format(instance_name))
+        operation = self._instances.reset(
+            project=self._project, zone=self._zone, instance=instance_name).execute()
+        wait_for(lambda: self._nested_wait_vm_running(operation['name']),
+            message="Restart {}".format(instance_name))
+        return True
+
+    def stop_vm(self, instance_name):
+        if self.is_vm_stopped(instance_name) or not self.does_vm_exist(instance_name):
+            self.logger.info("The {} instance is already stopped or doesn't exist, skip termination"
+               .format(instance_name))
+            return True
+
+        self.logger.info("Stoping Google Cloud instance {}".format(instance_name))
+        operation = self._instances.stop(
+            project=self._project, zone=self._zone, instance=instance_name).execute()
+        wait_for(lambda: self._nested_wait_vm_running(operation['name']),
+            message="Stop {}".format(instance_name))
+        return True
+
+    def start_vm(self, instance_name):
+        # This method starts an instance that was stopped using the using the
+        # instances().stop method.
+        if self.is_vm_running(instance_name) or not self.does_vm_exist(instance_name):
+            self.logger.info("The {} instance is already running or doesn't exists, skip starting"
+               .format(instance_name))
+            return True
+
+        self.logger.info("Starting Google Cloud instance {}".format(instance_name))
+        operation = self._instances.start(
+            project=self._project, zone=self._zone, instance=instance_name).execute()
+        wait_for(lambda: self._nested_wait_vm_running(operation['name']),
+            message="Start {}".format(instance_name))
+        return True
+
+    def clone_vm(self, source_name, vm_name):
+        raise NotImplementedError('clone_vm not implemented.')
+
+    # Get external IP (ephemeral)
+    def current_ip_address(self, vm_name):
+        return self.vm_status(vm_name)['natIP']
+
+    def deploy_template(self, template, *args, **kwargs):
+        raise NotImplementedError('deploy_template not implemented.')
+
+    def disconnect(self):
+        raise NotImplementedError('disconnect not implemented.')
+
+    def does_vm_exist(self, instance_name):
+        try:
+            self._find_instance_by_name(instance_name)
+            return True
+        except Exception:
```

Comment:
I believe this might mute more than we desire, it should be possible to pinpoint the "expected" exception class and use it here.